rest/src/main/java/org/jboss/pnc/rest/endpoint/BuildConfigurationSetEndpoint.java

```diff
@@ -186,6 +186,8 @@ public Response build(
     public Response addConfiguration(
             @ApiParam(value = "Build Configuration Set id", required = true) @PathParam("id") Integer id,
             BuildConfigurationRest buildConfig) {
+        if (buildConfigurationSetProvider.getSpecific(id).getBuildConfigurationIds().contains(buildConfig.getId()))
```

Comment:
Endpoints shouldn't contain any product related logic. I would prefer if you moved it to provider layer and add some tests for it.